### PR TITLE
Feat: Implement User Password change endpoint

### LINKED ISSUE #522: [FEATURE REQUEST]: Implement User password change endpoint
**Is your feature request related to a problem? Please describe.**
Currently this feature is not available on the project

**Describe how to intend to go about adding this feature**
Implement the endpoint from scratch

**Additional context**

### PR DESCRIPTION
**Pull request by oscar-ekeyekwu**
**Before submitting your PR for review**

- Run `npm run lint` to find errors in code syntax/format
- Run `npm run lint:fix` to fix all auto-fixable errors in source code and auto-format with prettier
- Ensure you fix any linting errors displayed after running any of the above commands

**What does this PR do?**

Fixes #522 (replace issue_number with the issue number this PR solves)

This endpoint allows a user to be able to change password

**description of Task to be completed?**

- [x] Create controller method
- [x] Create Password management util
- [x] Write validation rule for request body and params
- [x] Attach validation and authorization middleware to route
- [x] Update swagger documentation to include endpoint

**How should this be manually tested?**

Run the project locally and test at 
`localhost:5000/api/user/password/change-password`

**Any background context you want to provide?**

None

**Questions:**

No question

### CODE REVIEW

File: src/services/auth.js [hunk 1]
```diff
@@ -371,6 +372,45 @@ class UserService {
       user: user,
     };
   }
+
+  async changePassword(req) {
+    const { email, oldPassword, newPassword, confirmNewPassword } = req.body;
```
Comment (dave-ok) [inline src/services/auth.js]: The only values needed in the body are oldPassword and newPassword. We do not do password confirms on the server side. Also, the userId or identifier should be gotten from the token not the body.

File: src/services/auth.js [hunk 2]
```diff
@@ -371,6 +372,45 @@ class UserService {
...
+
+    try {
+      const user = await User.findOne({ email: email });
```
Comment (dave-ok) [inline src/services/auth.js:380]: This will then be a findbyId using the userId from token

File: src/services/auth.js [hunk 3]
```diff
@@ -371,6 +372,45 @@ class UserService {
...
+
+      if (newPassword !== confirmNewPassword) {
```
Comment (dave-ok) [inline src/services/auth.js]: This will no longer be needed

**Review by dave-ok — CHANGES_REQUESTED** [review]

**Review by dave-ok — COMMENTED** [review]
Awesome stuff! U just need a few tweaks

File: src/services/auth.js (hunk 2 above)
Comment (oscar-ekeyekwu) [reply src/services/auth.js:380]: at the moment the token does not contain this info
we will have to still use this email

File: src/services/auth.js (hunk 3 above)
Comment (oscar-ekeyekwu) [reply src/services/auth.js]: fixed

**Pull request merged**